### PR TITLE
Add new subsystem for file loading (#15)

### PR DESCRIPTION
**Pull request by mmmm1998**
* Support loading user files and game files with automatic path deduction - the class customers don't need worry about it
* Support blocking read/write operations and their async versions, which will executed in thread pool

### CODE REVIEW

File: src/common/filemanager.cpp [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+		run_worker(new_worker);
+		return result;
+	}
+
+private:
+	vector<ReportableWorker*> threads;
+
+	static std::chrono::milliseconds THREAD_LIVE_TIMEOUT;
+	static size_t ETERNAL_THREAD_COUNT;
+	static size_t START_THREAD_COUNT;
```
Comment (Svenny) [inline src/common/filemanager.cpp]: Constants may be expressed more explicitly like
`constexpr static size_t ETERNAL_THREAD_COUNT = 2;`

File: src/common/filemanager.cpp [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+path FileManager::gameDataPath() noexcept
+{
+	if constexpr (BuildConfig::kIsDeployBuild) {
+		// TODO: implement me
+		assert(false);
+		return std::filesystem::path();
+	} else {
+		return std::filesystem::current_path() / "data/game";
+	}
+}
```
Comment (Svenny) [inline src/common/filemanager.cpp]: These functions are technically not noexcept as `std::filesystem::current_path()` or `operator /` may throw.

File: src/common/filemanager.cpp (hunk 1 above)
Comment (mmmm1998) [reply src/common/filemanager.cpp]: Done

File: src/common/filemanager.cpp (hunk 2 above)
Comment (mmmm1998) [reply src/common/filemanager.cpp]: Done

**Review by Svenny — APPROVED** [review]
Approved.

**Pull request merged**